### PR TITLE
Added option to allow continiuing running build script if the build already exists

### PR DESCRIPTION
**Pull request by Chili-Man**

### CODE REVIEW

File: s3deploy.sh [hunk 1]
```diff
@@ -96,10 +96,15 @@ _create_git_tag() {
 
 # Exits from current build if the commit has already been tarballed in the
 # master branch. Only checks in the given date in 'YEAR/MONTH' format.
+# Takes the following args:
+#    $1, date : The date to check in 'YEAR/MONTH' format. Required.
+#    $2, dont_exit_if_build_exists : Whether to continue the script or not if the build already exists. Defaults to false; can bet set to any truthy value.
```
Comment (MSeal) [inline s3deploy.sh]: `bet` -> `be`

File: s3deploy.sh [hunk 2]
```diff
@@ -312,13 +324,14 @@ s3d_initialize() {
...
         # Install the aws cli tools
-        pip install $user_mode $ignore_installed awscli==1.7.11
+        pip install $user_mode $ignore_installed awscli==1.7.13
 
         # Update the path to access the aws executable
         if [ -z "$TRAVIS_PYTHON_VERSION" ]; then export PATH="$HOME/.local/bin/:$PATH"; fi
 
-        _check_build_exists $BUILD_DATE # Current month
-        _check_build_exists `date -u +%Y/%m --date '-1 month'` # Previous month
+        dont_exit_if_build_exists=$1
```
Comment (MSeal) [inline s3deploy.sh]: `$2`?

Comment (Chili-Man) [inline s3deploy.sh]: this is a seperate function

**Comment by MSeal** [comment]
:+1:

**Pull request merged**